readthedocs/api/v2/views/model_views.py

```diff
@@ -66,6 +66,49 @@ def render(self, data, accepted_media_type=None, renderer_context=None):
         return data.encode(self.charset)
 
 
+class DisableListEndpoint:
+
+    """
+    Helper to disable APIv2 listing endpoint.
+
+    We are disablng the listing endpoint because it could cause DOS without
+    using any type of filtering.
+
+    This class disables these endpoints except:
+
+     - version resource when passing ``?project__slug=``
+     - build resource when using ``?commit=``
+
+    All the other type of listings are disabled and return 409 CONFLICT with an
+    error message pointing the user to APIv3.
+    """
+
+    def list(self, *args, **kwargs):
+        disabled = True
+
+        # NOTE: keep list endpoint that specifies a resource
+        if any([
+                self.basename == 'version' and 'project__slug' in self.request.GET,
+                self.basename == 'build' and 'commit' in self.request.GET,
+        ]):
+            disabled = False
+
+        if not disabled:
+            return super().list(*args, **kwargs)
+
+        return Response(
+            {
+                'error': 'disabled',
+                'msg': (
+                    'List endpoint have been disabled: please provide a resource identifier. '
+                    'Take into account than APIv2 is planned to be deprecated soon. '
+                    'Please use APIv3: https://docs.readthedocs.io/page/api/v3.html'
+                )
+            },
+            status=status.HTTP_409_CONFLICT,
```

Comment:
I wish there was a better status code. I googled for this -- but we should probably comment here that there isn't a proper one to use.

Reply:
I will `410 GONE`, instead. Which seems to be exactly what we need.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/410

Reply:
Cool, we should try and get this released this week.